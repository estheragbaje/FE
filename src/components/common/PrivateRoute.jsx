/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { isLoggedIn } from '../../utils';

const PrivateRoute = ({ component: Component, ...rest }) => {
  const isSignedIn = isLoggedIn();
  if (isSignedIn) {
    return <Route {...rest} render={(props) => <Component {...props} />} />;
  }
  return <Redirect to="/login" />;
};

PrivateRoute.defaultProps = {
  exact: false,
};

PrivateRoute.propTypes = {
  exact: PropTypes.bool,
  component: PropTypes.shape({
    _ctor: PropTypes.func.isRequired,
  }).isRequired,
};

export default PrivateRoute;
