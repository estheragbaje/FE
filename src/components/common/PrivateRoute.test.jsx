import React from 'react';
import { shallow } from 'enzyme';

import PrivateRoute from './PrivateRoute';

describe('<PrivateRoute />', () => {
  it('renders the PrivateRoute component correctly', () => {
    const wrapper = shallow(<PrivateRoute />);
    expect(wrapper).toMatchSnapshot();
  });
});
