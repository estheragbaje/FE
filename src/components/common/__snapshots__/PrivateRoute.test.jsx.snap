// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> renders the PrivateRoute component correctly 1`] = `ShallowWrapper {}`;
